presto-main/src/main/java/com/facebook/presto/FullConnectorSession.java

```diff
@@ -109,6 +114,12 @@ public long getStartTime()
     }
 
     @Override
+    public <T> T getSystemProperty(String name, Class<T> type)
+    {
+       return type.cast(systemProperties.get(name));
```

Comment:
This should be `sessionPropertyManager.decodeProperty(name, systemProperties.get(name), type);` line in original proposal. Extract changes related to exposing system properties into separate patch.